apps/typegpu-docs/src/content/docs/guides/tgpu-cli.mdx

```diff
@@ -0,0 +1,521 @@
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader1.wgsl
+      - <FileDiff> shader1.ts </FileDiff>
+      - shader2.wgsl
+      - <FileDiff> shader2.ts </FileDiff>
+      - example
+          - shader3.wgsl
+</FileTree>
+</div>
+
+To specify a recursive search, use the `**` pattern:
+
+```bash
+tgpu-cli gen "path/to/**/*.wgsl"
+```
+
+This will generate TypeGPU definitions for all WGSL files inside the `path/to` directory and its subdirectories.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader1.wgsl
+      - <FileDiff> shader1.ts </FileDiff>
+      - example
+          - shader2.wgsl
+          - <FileDiff> shader2.ts </FileDiff>
+</FileTree>
+</div>
+
+### Specify the output path with the `--output` option
+
+By default, the generated TypeScript files are placed in the same directory as the input shaders.
+You can specify a different output path using the `--output` option:
+
+```bash
+tgpu-cli gen "path/to/shader.wgsl" --output "different/path/output.ts"
+```
+
+This will generate the TypeGPU definitions in the `different/path` directory with the `output.ts` filename.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader.wgsl
+  - <FileDiff> different/path </FileDiff>
+      - <FileDiff> output.ts </FileDiff>
+</FileTree>
+</div>
+
+It also supports glob patterns for specifying the output path:
+
+```bash
+tgpu-cli gen "path/to/*.wgsl" --output "output/*.ts"
+```
+
+This will generate the TypeGPU definitions in the `output` directory with the same names as the shaders, but with the `.ts` extension.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader1.wgsl
+      - shader2.wgsl
+      - example
+          - shader3.wgsl
+  - <FileDiff> output </FileDiff>
+      - <FileDiff> shader1.ts </FileDiff>
+      - <FileDiff> shader2.ts </FileDiff>
+</FileTree>
+</div>
+
+:::caution
+If you use a glob pattern for the input path, specifically `**`, make sure to use the `**` pattern
+in the output path as well to preserve the directory structure (*or make sure no conflicts will happen*).
+If you don't, all generated files will be placed in the same directory which may lead to name conflicts.
+
+<FileTree>
+
+- src
+  - example
+    - shader1.wgsl
+  - shader1.wgsl
+  - shader2.wgsl
+
+</FileTree>
+
+If you run:
+
+```bash
+tgpu-cli gen "src/**/*.wgsl" --output "output/*.ts"
+```
+
+The script will fail because all generated files will be placed in the `output` directory, and there will
+be a name conflict as both `shader1.wgsl` and `example/shader1.wgsl` will try to generate `output/shader1.ts`.
+
+<div class="relative">
+<FileTree>
+  - src
+      - example
+          - shader1.wgsl
+      - shader1.wgsl
+      - shader2.wgsl
+  - <FileDiff> output </FileDiff>
+      - <FileDiff> shader1.ts </FileDiff> \<\-\- name conflict
+      - <FileDiff> shader1.ts </FileDiff> \<\-\- name conflict
+      - <FileDiff> shader2.ts </FileDiff>
+</FileTree>
+</div>
+:::
```

Comment:
after this I would add an example of how to fix this. showing what directory structure `tgpu-cli gen "src/**/*.wgsl" --output "output/**/*.ts"` generates instead